src/sync/mod.rs

```diff
@@ -9,3 +9,4 @@ mod atomic_option;
 mod ms_queue;
 mod treiber_stack;
 mod seg_queue;
+pub mod deque;
```

Comment:
If only we had inherent associated types!

Exporting a module is fine. However, please rename to something like `cl_deque` or `chase_lev_deque`. For the moment, I'm trying to keep fairly specific about the algorithm being implemented. Later on, we may have aliases for our "preferred" queues, deques, etc.